test/Elastic.Apm.Tests/ConfigTests.cs

```diff
@@ -162,31 +162,35 @@ public void SetDebugLogLevelTest()
 		{
 			var agent = new ApmAgent(new TestAgentComponents("Debug"));
 			agent.ConfigurationReader.LogLevel.Should().Be(LogLevel.Debug);
-			agent.Logger.Level.Should().Be(LogLevel.Debug);
+			agent.Logger.IsEnabled(LogLevel.Debug).Should().BeTrue();
```

Comment:
Would something like this method make sense and calling it from the respective `SetXDebugLogLevelTest`?

```
private static void TestAllLevelsWithEnabledLevel(IApmAgent agent, LogLevel enabledLevel)
{
	agent.Logger.IsEnabled(enabledLevel).Should().BeTrue();
	foreach (LogLevel enumValue in Enum.GetValues(typeof(LogLevel)))
	{
		if(enabledLevel <= enumValue)
			agent.Logger.IsEnabled(enumValue).Should().BeTrue();
		else
			agent.Logger.IsEnabled(enumValue).Should().BeFalse();
	}
}
```

And in this case the Debug test would just call `TestAllLevelsWithEnabledLevel(agent, LogLevel.Debug);` - and some for the other tests.

Reply:
The valid point, will refactor unit tests